File: Samples/Layout/HelloWorld/C#/HelloWorld4.cs
using System;

using BitMiracle.Docotic.Pdf.Layout;

namespace BitMiracle.Docotic.Pdf.Samples
{
    class HelloWorld4
    {
        public static void CreatePdf()
        {
            PdfDocumentBuilder.Create().Generate("hello.pdf", BuildDocument);
        }

        static void BuildDocument(Document doc)
        {
            doc.Typography(t =>
            {
                t.Document = doc.TextStyleWithFont(SystemFont.Family("Calibri"));
            });
            doc.Pages(BuildPages);
        }

        static void BuildPages(PageLayout pages)
        {
            pages.Size(PdfPaperSize.A6);
            pages.Margin(10);

            BuildPagesHeader(pages.Header());
            BuildPagesFooter(pages.Footer());

            BuildTextContent(pages.Content());
        }

        static void BuildTextContent(LayoutContainer content)
        {
            var colorAccented = new PdfRgbColor(56, 194, 10);
            var styleAccented = TextStyle.Parent.FontColor(colorAccented);

            content.Text(t =>
            {
                t.Span("Hello, ").Style(styleAccented);
                t.Span("world!").Style(styleAccented.Underline());
            });
        }

        static void BuildPagesHeader(LayoutContainer header)
        {
            header.TextStyle(TextStyle.Parent.FontSize(8))
                .AlignRight()
                .Text(t =>
                {
                    t.Line($"Created by: {Environment.UserName}");
                    t.Line($"Date: {DateTime.Now}");
                });
        }

        static void BuildPagesFooter(LayoutContainer footer)
        {
            footer.Height(20)
                .Background(new PdfGrayColor(95))
                .AlignCenter()
                .Text(t => t.CurrentPageNumber());
        }
    }
}

File: Samples/Layout/HelloWorld/VB.NET/HelloWorld4.vb
Imports BitMiracle.Docotic.Pdf.Layout

Namespace BitMiracle.Docotic.Pdf.Samples
    Class HelloWorld4
        Public Shared Sub CreatePdf()
            PdfDocumentBuilder.Create().Generate("hello.pdf", AddressOf BuildDocument)
        End Sub

        Private Shared Sub BuildDocument(doc As Document)
            doc.Typography(Sub(t) t.Document = doc.TextStyleWithFont(SystemFont.Family("Calibri")))
            doc.Pages(AddressOf BuildPages)
        End Sub

        Private Shared Sub BuildPages(pages As PageLayout)
            pages.Size(PdfPaperSize.A6)
            pages.Margin(10)
            BuildPagesHeader(pages.Header())
            BuildPagesFooter(pages.Footer())
            BuildTextContent(pages.Content())
        End Sub

        Private Shared Sub BuildTextContent(content As LayoutContainer)
            Dim colorAccented = New PdfRgbColor(56, 194, 10)
            Dim styleAccented = TextStyle.Parent.FontColor(colorAccented)
            content.Text(
                Sub(t)
                    t.Span("Hello, ").Style(styleAccented)
                    t.Span("world!").Style(styleAccented.Underline())
                End Sub)
        End Sub

        Private Shared Sub BuildPagesHeader(header As LayoutContainer)
            header.TextStyle(TextStyle.Parent.FontSize(8)).
                AlignRight().Text(
                    Sub(t)
                        t.Line($"Created by: {Environment.UserName}")
                        t.Line($"Date: {DateTime.Now}")
                    End Sub)
        End Sub

        Private Shared Sub BuildPagesFooter(footer As LayoutContainer)
            footer.Height(20).
                Background(New PdfGrayColor(95)).
                AlignCenter().
                Text(Sub(t) t.CurrentPageNumber())
        End Sub
    End Class
End Namespace
